apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py

```diff
@@ -0,0 +1,366 @@
+                      (common.server, common.server_value)])
+      return ['Missing Metadata-Flavor:Google header.']
+    return self.app(environ, start_response)
+
+# Collection whose content is accessed via an index value in the relative URL.
+class IndexedList:
+  def __init__(self, items):
+    self._items = items
+
+  def __iter__(self):
+    return self._items.__iter__()
+
+  def __len__(self):
+    return self._items.__len__()
+
+  def __delitem__(self, key):
+    self._items.__delitem__(key)
+
+  def __getitem__(self, key):
+    return self._items.__getitem__(key)
+  
+  def __setitem__(self, key, value):
+    self._items.__setitem__(key, value)
+
+  def formatContents(self):
+    return ''.join(unicode(i) + u'/\n' for i in range(len(self._items)))
+
+class MetadataJSONEncoder(json.JSONEncoder):
+  def ensureOrderedDict(self, o):
+    if isinstance(o, OrderedDict):
+      return self.fixupOrderedDict(o)
+
+    return o
+
+  def fixupOrderedDict(self, d):
+    result = OrderedDict()
+    for key in d:
+      if excludeKeyInRecursiveGet(key):
+        continue
+
+      if key.endswith(u'@developer.gserviceaccount.com'):
+        newKey = key
+      else:
+        newKey = common.convertToCamelCase(key)
+
+      if (d[key] == numeric_project_id_token):
+        result[newKey] = numeric_project_id
+      else:
+        result[newKey] = self.default(d[key])
+
+    return result
+
+  def encode(self, o):
+    return json.JSONEncoder.encode(self, self.ensureOrderedDict(o))
+
+  def default(self, o):
+    if isinstance(o, IndexedList):
+      return [self.default(item) for item in o]
+    
+    return self.ensureOrderedDict(o)
+
+not_found_error = formatResponse(u'Error: Not found', 404, common.content_type_html)
+bad_request_error = formatResponse(u'Error: bad request', 400, common.content_type_html)
+
+all_data = {}
+def loadData():
+  global all_data
+  with open(os.path.join(os.path.dirname(os.path.realpath(__file__)), 'initial_data.json')) as initial_data_file:
+    all_data = json.load(initial_data_file, object_pairs_hook=OrderedDict)
+    indexed_lists = [
+      u'instance/disks',
+      u'instance/licenses',
+      u'instance/network-interfaces',
+      u'instance/network-interfaces/*/access-configs',
+      u'instance/network-interfaces/*/forwarded-ips',
+      u'instance/network-interfaces/*/ip-aliases'
+    ]
+    for indexed_list in indexed_lists:
+      components = indexed_list.split('/')
+      def process(data, i):
+        component = components[i]
+        if component == '*':
+          for item in data:
+            process(item, i + 1)
+        elif i == len(components) - 1:
+          data[component] = IndexedList(data[component])
+        else:
+          process(data[component], i + 1)
+
+      process(all_data, 0)
+
+def getRawDataAt(path):
+  result = all_data
+  for component in path.split('/'):
+    if len(component) == 0:
+      continue
+
+    if isinstance(result, IndexedList):
```

Comment:
these type checks everywhere are a bit of a code smell...we don't have to block this on that, but it generally means that something is wrong in my experience. Especially since you control the result type! you set it when you load the data...